api/fma/v1alpha1/launcherpoolpolicy_types.go

```diff
@@ -0,0 +1,220 @@
+
+// LauncherPoolForNodeType defines launcher count for a class of nodes.
+type LauncherPoolForNodeType struct {
+	// Selector describes the hardware characteristics of target nodes.
+	// +kubebuilder:validation:Required
+	// Introduce an EnhancedNodeSelector that supports combining label-based
+	// matching with resource field conditions.
+	// For example:
+	// enhancedNodeSelector:
+	//  # 1. Label selector (compatible with existing metav1.LabelSelector)
+	//  labelSelector:
+	//    matchLabels:
+	//      nvidia.com/gpu.family: ada-lovelace
+	//    matchExpressions:
+	//      - key: node.kubernetes.io/instance-type
+	//        operator: In
+	//        values: ["gx3-48x240x2l40s", "gx3-96x480x4l40s"]
+	//
+	//  # 2. Resource condition selector (new capability)
+	//  resourceRequirements:
+	//    allocatable:
+	//      cpu:
+	//        min: "16"
+	//        max: "64"
+	//      memory:
+	//        min: "128Gi"
+	//        max: "512Gi"
+	//      gpus:
+	//        nvidia.com/gpu:
+	//          min: "2"
+	//	        max: "8"
+	//  acceleratorSelector:
+	//    type: "nvidia.com/gpu"
+	//    memory: "48Gi"
+	//    count: 4
+	EnhancedNodeSelector EnhancedNodeSelector `json:"enhancedNodeSelector"`
+
+	// PerAcceleratorCount defines pre-configuration quantities for each accelerator type
+	PerAcceleratorCount []PerAcceleratorCount `json:"perAcceleratorCount"`
+}
+
+// PerAcceleratorCount defines configuration for specific accelerators
+type PerAcceleratorCount struct {
+	// AcceleratorSelector accelerator selector
+	AcceleratorSelector AcceleratorSelector `json:"acceleratorSelector"`
+
+	// CountForLauncher is the total number of launcher for each LauncherConfig
+	// to maintain on each matching node per accelerator.
+	// If two different counts are specified for the same (Node, Accelerator, LauncherConfig),
+	// the higher count is used and will be populated into LauncherPoolPolicyStatus.
+	// If no CountForLauncher applies to a given (Node, Accelerator, LauncherConfig), this Node
+	// will be ignored for this LauncherConfig.
+	// +kubebuilder:validation:MinItems=1
+	CountForLauncher []CountForLauncher `json:"countForLauncher"`
+}
+
+// ResourceRequirements defines resource requirements for a node.
+type ResourceRequirements struct {
+	// Allocatable defines the allocatable resources for a node with min/max ranges.	// +kubebuilder:validation:Required
+	Allocatable *ResourceRanges `json:"allocatable,omitempty"`
+}
+
+// ResourceRanges defines min/max ranges for various resources.
+type ResourceRanges struct {
+	// CPU defines the CPU resource range requirement.
+	// +kubebuilder:validation:Optional
+	CPU *ResourceRange `json:"cpu,omitempty"`
+
+	// Memory defines the memory resource range requirement.
+	// +kubebuilder:validation:Optional
+	Memory *ResourceRange `json:"memory,omitempty"`
+
+	// GPUs defines the GPU resource range requirements keyed by GPU type.
+	// +kubebuilder:validation:Optional
+	GPUs map[string]*ResourceRange `json:"gpus,omitempty"`
```

Comment:
What about TPUs? Maybe call this field "accelerators"?

Reply:
fine.